shards/shards.go

```diff
@@ -367,51 +428,34 @@ func (ss *shardedSearcher) Search(ctx context.Context, q query.Q, opts *zoekt.Se
 		feeder <- s
 	}
 	close(feeder)
+	g := errgroup.Group{}
```

Comment:
should we use withcontext so that if searchOneShard returns an error we cancel the context?

Reply:
@stefanhengl wdyt?

Reply:
yes!